pkg/spanconfig/spanconfig.go

```diff
@@ -33,35 +36,72 @@ type KVAccessor interface {
 	// divvying up an existing span into multiple others with distinct configs,
 	// callers are to issue a delete for the previous span and upserts for the
 	// new ones.
-	UpdateSpanConfigEntries(ctx context.Context, toDelete []roachpb.Span, toUpsert []roachpb.SpanConfigEntry) error
+	UpdateSpanConfigEntries(
+		ctx context.Context,
+		toDelete []roachpb.Span,
+		toUpsert []roachpb.SpanConfigEntry,
+	) error
+}
+
+// KVSubscriber presents a consistent[1] snapshot of a StoreReader that's
+// incrementally maintained with changes made to the global span configurations
+// state (system.span_configurations). The maintenance happens transparently;
+// callers can subscribe to learn about what key spans may have seen a
+// configuration change. After learning about a span update through a callback
+// invocation, callers can consult the embedded StoreReader to retrieve an
+// up-to-date[2] config for the updated span. The callback is called in a single
+// goroutine; it should avoid doing long-running/blocking work.
+//
+// When the callback is first installed (there can only be one), it's invoked
```

Comment:
this "there can only be one" is surprising to me. What's the logic there?

Reply:
It's very easy to extend this to multiple callbacks, we just didn't need to. Keeping it to a single callback made lazy-binding + initialization of handlers slightly easier.